### PR TITLE
NLB-4374: Add validation of set_real_ip_from  directive into crossplane

### PR DESCRIPTION
**Pull request by xynicole**
### Proposed changes

Add validation of set_real_ip_from  directive into crossplane.  
Refer https://github.com/nginx/nginx/blob/ef96f5835468ff8d40df29b0ddbc04ec1e5e1582/src/mail/ngx_mail_realip_module.c#L30 

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/nginx-go-crossplane/blob/main/CONTRIBUTING.md) document
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] I have updated any relevant documentation ([`README.md`](https://github.com/nginxinc/nginx-go-crossplane/blob/main/README.md))

### CODE REVIEW

**Review by dakshinai — APPROVED** [review]

**Review by ornj — APPROVED** [review]
[lgtm](https://github.com/nginx/nginx/blob/master/src/mail/ngx_mail_realip_module.c#L30)

File: analyze.go [hunk 1]
```diff
@@ -1595,6 +1595,7 @@ var directives = map[string][]uint{
 	"set_real_ip_from": {
 		ngxHTTPMainConf | ngxHTTPSrvConf | ngxHTTPLocConf | ngxConfTake1,
 		ngxStreamMainConf | ngxStreamSrvConf | ngxConfTake1,
+		ngxMailMainConf | ngxMailSrvConf | ngxConfTake1,
```
Comment (puneetsarna) [inline analyze.go:1598]: is this something that is supported? have you tried this out?

https://nginx.org/en/docs/stream/ngx_stream_realip_module.html based on this it is just for stream and server but does not mention `server` under `mail`?

Comment (xynicole) [reply analyze.go:1598]: https://nginx.org/en/docs/mail/ngx_mail_realip_module.html  and https://github.com/nginx/nginx/blob/ef96f5835468ff8d40df29b0ddbc04ec1e5e1582/src/mail/ngx_mail_realip_module.c#L30 says it `"set_real_ip_from":`also for mail and server

Comment (puneetsarna) [reply analyze.go:1598]: aah. I see it. it's the mail module. Thanks for sharing this @xynicole

**Pull request merged**